src/azure_extension.cpp

```diff
@@ -1,81 +1,147 @@
 #define DUCKDB_EXTENSION_MAIN
 
 #include "azure_extension.hpp"
+
 #include "duckdb.hpp"
 #include "duckdb/common/exception.hpp"
-#include "duckdb/common/string_util.hpp"
-#include "duckdb/function/scalar_function.hpp"
 #include "duckdb/common/file_opener.hpp"
+#include "duckdb/common/string_util.hpp"
 #include "duckdb/function/scalar/string_functions.hpp"
-
+#include "duckdb/function/scalar_function.hpp"
+#include "duckdb/main/extension_util.hpp"
+#include <azure/storage/blobs.hpp>
+#include <azure/identity/default_azure_credential.hpp>
+#include <azure/identity/chained_token_credential.hpp>
+#include <azure/identity/environment_credential.hpp>
+#include <azure/identity/managed_identity_credential.hpp>
+#include <azure/identity/azure_cli_credential.hpp>
 #include <duckdb/parser/parsed_data/create_scalar_function_info.hpp>
-
 #include <iostream>
 
-#include <azure/storage/blobs.hpp>
-
 namespace duckdb {
 
-BlobClientWrapper::BlobClientWrapper(AzureAuthentication auth, const string& path) {
-	auto container_client = Azure::Storage::Blobs::BlobContainerClient::CreateFromConnectionString(auth.connection_string, auth.container);
-	blob_client = make_uniq<Azure::Storage::Blobs::BlockBlobClient>(container_client.GetBlockBlobClient(path));
+static Azure::Identity::ChainedTokenCredential::Sources CreateCredentialChainFromSetting(string& credential_chain) {
+	auto chain_list = StringUtil::Split(credential_chain, ';');
```

Comment:
yea i wanted to do a List type here, but there's no parser support for lists in settings I think. Since I want this feature to target 0.9.1 I did it like this.